src/components/NavButton.tsx

```diff
@@ -6,14 +6,18 @@ export interface NavButtonProps {
     label: string,
     targetUrl: string,
     icon: ReactNode
+    setNavOpened?: React.Dispatch<React.SetStateAction<boolean>>
```

Comment:
Since we know that setNavOpened will be provided by the parent, we do not need to set the prop to be optional. Setting setNavOpened to optional types the value as either the Dispatch<SetStateAction<boolean>> or undefined, which will lead to more code being written to checking if the value is undefined or not.